src/components/bottom-sheet-v2/bottom-sheet-modal/BottomSheetModal.tsx

```diff
@@ -0,0 +1,127 @@
+import React, {PropsWithChildren, useCallback, useState} from 'react';
+import {
+  BottomSheetModal as GorhomBottomSheetModal,
+  BottomSheetBackdrop,
+  BottomSheetFooter,
+  BottomSheetScrollView,
+} from '@gorhom/bottom-sheet';
+import {BottomSheetHeader} from '../BottomSheetHeader';
+import {SvgProps} from 'react-native-svg';
+import {Platform, useWindowDimensions, View} from 'react-native';
+import {useBottomSheetStyles} from '../use-bottom-sheet-styles';
+import {ReduceMotion} from 'react-native-reanimated';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+import {useThemeContext} from '@atb/theme';
+
+type BottomSheetModalProps = PropsWithChildren<{
+  BottomSheetModalRef: React.RefObject<GorhomBottomSheetModal | null>;
+  heading?: string;
+  subText?: string;
+  logoUrl?: string;
+  rightIcon?: (props: SvgProps) => React.JSX.Element;
+  rightIconText?: string;
+  headerNode?: React.ReactNode;
+  snapPoints?: Array<string | number>;
+  enableDynamicSizing?: boolean;
+  keyboardBehavior?: 'extend' | 'interactive' | 'fillParent';
+  closeCallback?: () => void;
+  footer?: React.ReactNode;
+}>;
+export const BottomSheetModal = ({
+  children,
+  BottomSheetModalRef,
+  heading,
+  subText,
+  logoUrl,
+  rightIcon,
+  rightIconText,
+  headerNode,
+  snapPoints,
+  enableDynamicSizing = true,
+  keyboardBehavior = Platform.OS === 'ios' ? 'interactive' : 'extend',
+  closeCallback,
+  footer,
+}: BottomSheetModalProps) => {
+  const styles = useBottomSheetStyles();
+  const {height: screenHeight} = useWindowDimensions();
+  const {top: safeAreaTop, bottom: safeAreaBottom} = useSafeAreaInsets();
+  const [footerHeight, setFooterHeight] = useState(0);
+  const {theme} = useThemeContext();
+
+  const renderBackdrop = useCallback(
```

Comment:
For this and `renderFooter`, can we use actual react components instead of useCallbacks, similar to BottomSheetHeader?